### ISSUE TITLE
aws_iam_role.force_detach_policies no longer works

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#2279]
_This issue was originally opened by @aleybovich as hashicorp/terraform#16637. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Reproduce: 
1. Create an aws role with terraform

 ```
resource "aws_iam_role" "sample-instance-role" {
    name               = "sample-instance-role"
    path               = "/"
    force_detach_policies = true
    ... other stuff ...
}
```

2. Manually add another policy to that role (e.g. via aws console)
3. run `terraform destroy`

You get an error `sample-instance-role: DeleteConflict: Cannot delete entity, must detach all policies first.`

`force_detach_policies` feature has been added a few months ago specifically for this purpose - it detaches all policies that weer added outside of terraform so that the role can be deleted. This used to work briefly in one or two versions but it no longer does in 0.10.7

Documentation: https://www.terraform.io/docs/providers/aws/r/iam_role.html#force_detach_policies

**Comment by atsushi-ishibashi** [comment]
@aleybovich There is a possibility that policies is truncated because a lot of policies are attached.
Could you share how many policies are attached?
https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#ListAttachedRolePoliciesOutput

**Comment by aleybovich** [comment]
There are five policies that get attached to all aws roles automatically
via Turbot, after those roles are added. That was the reason that I've
filed the initial change request a year or so ago and it was implemented.
It used to work for a bit but now no longer does.

**Comment by atsushi-ishibashi** [comment]
@aleybovich 
https://docs.aws.amazon.com/IAM/latest/APIReference/API_ListRolePolicies.html
> Lists the names of the inline policies that are embedded in the specified IAM role. 
> An IAM role can also have managed policies attached to it. To list the managed policies that are attached to a role, use ListAttachedRolePolicies. 

Probably inline policies are still attached. For `force_detach_policies`, `ListAttachedRolePolicies` is used.

**Comment by aleybovich** [comment]
@atsushi-ishibashi - the only inline policy that role has is the one I add in the same terraform template; all Turbot policies are attached.

**Comment by atsushi-ishibashi** [comment]
@aleybovich I submitted PR to fix this👍

**Comment by aleybovich** [comment]
Thank you! Whats the usual turnaround for a pull request like this one?

**Comment by atsushi-ishibashi** [comment]
Probably...
open issue -> anyone submit PR
or
submit PR with details when you find and solve by yourself.

Refer [CONTRIBUTING.md](https://github.com/terraform-providers/terraform-provider-aws/blob/master/.github/CONTRIBUTING.md)

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
